lib/check_open_telemetry_config.rb

```diff
@@ -0,0 +1,83 @@
+require 'opentelemetry/sdk'
+require 'opentelemetry/exporter/otlp'
+require 'opentelemetry/instrumentation/all'
+
+module Check
```

Comment:
This logic is kind of complicated, but there's a lot of room for error in the way that Open Telemetry uses environment variables for configuration. I'm trying to provide some guiderails around this, and also do so in a way that lets us unit test and control the initialization.